src/firmware/app/primitives/move_helper.c

```diff
@@ -157,55 +64,97 @@ void app_move_helper_start(void* void_state_ptr, FirmwareWorld_t* world,
+    const float destination_x           = move_position_params.destination.x_meters;
+    const float destination_y           = move_position_params.destination.y_meters;
+    const float destination_orientation = final_angle;
+    const float speed_at_dest_m_per_s =
+        move_position_params.final_speed_meters_per_second;
+    state->move_slow = move_position_params.slow;
 
     const FirmwareRobot_t* robot = app_firmware_world_getRobot(world);
 
-    float dx = state->destination[0] - app_firmware_robot_getPositionX(robot);
-    float dy = state->destination[1] - app_firmware_robot_getPositionY(robot);
-    // Add a small number to avoid division by zero
-    float total_disp    = sqrtf(dx * dx + dy * dy) + 1e-6f;
-    state->major_vec[0] = dx / total_disp;
-    state->major_vec[1] = dy / total_disp;
-    state->minor_vec[0] = state->major_vec[0];
-    state->minor_vec[1] = state->major_vec[1];
-    rotate(state->minor_vec, P_PI / 2);
-
-    // pick the wheel axis that will be used for faster movement
-    state->optimal_wheel_axes_index = choose_wheel_axis(
-        dx, dy, app_firmware_robot_getOrientation(robot), state->destination[2]);
+    const float current_x           = app_firmware_robot_getPositionX(robot);
+    const float current_y           = app_firmware_robot_getPositionY(robot);
+    const float current_orientation = app_firmware_robot_getOrientation(robot);
+    const float current_speed       = app_firmware_robot_getSpeedLinear(robot);
+
+    // Plan a trajectory to move to the target position/orientation
+    FirmwareRobotPathParameters_t path_parameters = {
+        .path = {.x = {.coefficients = {0, 0, destination_x - current_x, current_x}},
+                 .y = {.coefficients = {0, 0, destination_y - current_y, current_y}}},
+        .orientation_profile = {.coefficients = {0, 0,
+                                                 fmodf(destination_orientation -
+                                                           current_orientation,
+                                                       (float)M_PI),
+                                                 current_orientation}},
+        .t_start             = 0,
+        .t_end               = 1.0f,
+        .num_elements        = 10,
+        .max_allowable_linear_acceleration =
+            (float)ROBOT_MAX_ACCELERATION_METERS_PER_SECOND_SQUARED,
+        .max_allowable_linear_speed = (float)ROBOT_MAX_SPEED_METERS_PER_SECOND,
+        .max_allowable_angular_acceleration =
+            (float)ROBOT_MAX_ANG_ACCELERATION_RAD_PER_SECOND_SQUARED,
+        .max_allowable_angular_speed = (float)ROBOT_MAX_ANG_SPEED_RAD_PER_SECOND,
+        .initial_linear_speed        = current_speed,
+        .final_linear_speed          = speed_at_dest_m_per_s};
+    state->num_trajectory_elems = path_parameters.num_elements;
+    app_trajectory_planner_generateConstantParameterizationPositionTrajectory(
+        path_parameters, &(state->position_trajectory));
+
```

Comment:
This looks pretty garbled by clash format. Maybe we should turn it off and manually format